src/main/java/teammates/ui/webapi/action/UpdateStudentAction.java

```diff
@@ -0,0 +1,99 @@
+package teammates.ui.webapi.action;
+
+import java.util.Arrays;
+
+import org.apache.http.HttpStatus;
+
+import teammates.common.datatransfer.attributes.InstructorAttributes;
+import teammates.common.datatransfer.attributes.StudentAttributes;
+import teammates.common.exception.EnrollException;
+import teammates.common.exception.EntityAlreadyExistsException;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.common.exception.TeammatesException;
+import teammates.common.exception.UnauthorizedAccessException;
+import teammates.common.util.Const;
+import teammates.common.util.EmailWrapper;
+import teammates.common.util.Logger;
+import teammates.logic.api.EmailGenerator;
+import teammates.ui.webapi.request.StudentUpdateRequest;
+
+/**
+ * Action: Edits details of a student in a course.
+ */
+public class UpdateStudentAction extends Action {
+    private static final Logger log = Logger.getLogger();
+
+    @Override
+    protected AuthType getMinAuthLevel() {
+        return authType.LOGGED_IN;
+    }
+
+    @Override
+    public void checkSpecificAccessControl() {
+        if (!userInfo.isInstructor) {
+            throw new UnauthorizedAccessException("Instructor privilege is required to access this resource.");
+        }
+        String courseId = getRequestParamValue(Const.ParamsNames.COURSE_ID);
+
+        InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, userInfo.id);
+        gateKeeper.verifyAccessible(
+                instructor, logic.getCourse(courseId), Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_STUDENT);
+    }
+
+    @Override
+    public ActionResult execute() {
+        String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
+        String studentEmail = getNonNullRequestParamValue(Const.ParamsNames.STUDENT_EMAIL);
+
+        StudentAttributes student = logic.getStudentForEmail(courseId, studentEmail);
+        if (student == null) {
+            return new JsonResult(Const.StatusMessages.STUDENT_NOT_FOUND_FOR_EDIT, HttpStatus.SC_NOT_FOUND);
+        }
+
+        StudentUpdateRequest updateRequest = getAndValidateRequestBody(StudentUpdateRequest.class);
+        try {
+            logic.validateSectionsAndTeams(Arrays.asList(student), student.course);
+            logic.validateSectionsAndTeams(Arrays.asList(student), student.course);
```

Comment:
Do you need two of the same call?

Reply:
Oh no! one of them is supposed to be `validateTeams` (have updated)